File: HackerRank/C30daysofCode/q7.cpp
#include <iostream>
#include <string>
using namespace std;


int main() {
    /* Enter your code here. Read input from STDIN. Print output to STDOUT */
    int n =0;
    cin>>n;
    cin.ignore();
    string s1,sEven,sOdd;
    //cout<<" N is "<<n;
    for(int i =0;i<n;++i)
    {
        s1="";
      getline(cin,s1);
      //cout<<"read string is "<<s1;
      sEven ="";
      sOdd="";
      for(int i =0;i<s1.length();++i)
      {
          if(i%2==0)
            sEven.push_back(s1.at(i));
          else
            sOdd.push_back(s1.at(i));
      }
      cout<<sEven<<" "<<sOdd<<endl;
    }


    return 0;
}

File: HackerRank/C30daysofCode/q7.java
import java.io.*;
import java.util.*;

public class Solution {

    public static void main(String[] args) {
        /* Enter your code here. Read input from STDIN. Print output to STDOUT. Your class should be named Solution. */

        int n=0;
        String s1,sEven,sOdd;

        Scanner scanObj = new Scanner(System.in);
        n=scanObj.nextInt();
        scanObj.nextLine(); //trim \n 
        s1="";
        for(int i =0;i<n;++i)
        {
            s1= scanObj.nextLine();
            sEven="";
            sOdd="";
            for(int j=0;j<s1.length();++j)
            {
                if(j%2==0)
                sEven=sEven+(s1.charAt(j));
                else
                sOdd+=(s1.charAt(j));
            }
            System.out.println(sEven+" "+sOdd);
        }
    }
}